src/org/wikipedia/Wiki.java

```diff
@@ -4919,8 +4918,8 @@ public boolean isWatched(String title) throws IOException, CredentialNotFoundExc
                 boolean iscat = namespace(member) == CATEGORY_NAMESPACE;
                 if (maxdepth > 0 && iscat && !visitedcategories.contains(member))
                 {
-                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
                     visitedcategories.add(member);
+                    String[] categoryMembers = getCategoryMembers(member, --maxdepth, visitedcategories, ns);
```

Comment:
I admit, my background isn’t in Java, but isn’t the prefix-operator `--` equivalent?